File: src/main/java/com/equalinformation/poc/akka/java/track1/future/OrderAggregateActor.java
package com.equalinformation.poc.akka.java.track1.future;

import akka.actor.UntypedActor;
import com.equalinformation.poc.akka.java.track1.future.messages.OrderHistory;

/**
 * Created by bpupadhyaya on 2/12/16.
 */
public class OrderAggregateActor extends UntypedActor {

    @Override
    public void onReceive(Object message) throws Exception {
        if (message instanceof OrderHistory) {
            OrderHistory orderHistory = (OrderHistory) message;
            System.out.println("Order History -> " + orderHistory.getOrder()
                    + "\n" + orderHistory.getAddress());
        }
    }
}

File: src/main/scala/com/equalinformation/poc/akka/scala/track1/future/OrderAggregateActor.scala
package com.equalinformation.poc.akka.scala.track1.future

import akka.actor.Actor

/**
  * Created by bpupadhyaya on 2/12/16.
  */
class OrderAggregateActor extends Actor {
  def receive = {
    case orderHistory: OrderHistory =>
      println(orderHistory.toString())
  }
}
